### PR TITLE
[PVR] PVR addon call guards

### PR DESCRIPTION
**Pull request by ksooo**
This is something I had in the pipeline for a long time and was tested by me over months in my living room setup. It has fixed for me occasional crashes on Kodi shutdown, addon update, addon uninstall, ... - all caused by destroying the addon instance while it was still actively used or by addons calling back into Kodi while PVR component was no longer started.

This PR contains two commits:
1) Wait for return of all in-progress addon calls before destroying the addon instance.
2) Ignore addon callbacks when PVR manager is stopping/stopped.

I hope the code is pretty self-explaining.

@phunkyfish could you review the changes, please?

### CODE REVIEW

File: xbmc/pvr/addons/PVRClient.cpp [hunk 1]
```diff
@@ -1647,299 +1657,313 @@ int CPVRClient::GetPriority() const
...
+                strConnectionString, client->Name(), prevState, newState);
 
-  client->SetConnectionState(newState);
+    client->SetConnectionState(newState);
 
-  std::string msg;
-  if (strMessage != nullptr)
-    msg = strMessage;
+    std::string msg;
+    if (strMessage != nullptr)
```
Comment (phunkyfish) [inline xbmc/pvr/addons/PVRClient.cpp:1943]: `If (!strMessage)` maybe?

**Review by phunkyfish — COMMENTED** [review]
Nice, just one minor as far as I can see. I can runtime test it at the weekend. Might be good to see how it works against a few different PVR add-ons that I have and approve afterwards? WDYT?

**Comment by ksooo** [comment]
> Might be good to see how it works against a few different PVR add-ons that I have and approve afterwards? WDYT?

 Sounds good.

File: xbmc/pvr/addons/PVRClient.cpp (hunk 1 above)
Comment (ksooo) [reply xbmc/pvr/addons/PVRClient.cpp:1943]: This was also in the original code, but I can change that.

**Comment by ksooo** [comment]
jenkins build this please

**Review by phunkyfish — APPROVED** [review]
Got to build and test this today. Works well.

**Pull request merged**